Failure in cmdLineTester_DataHelperTests test under forceAOT

Failure link
------------

https://140-211-168-230-openstack.osuosl.org/artifactory/webapp/builds/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/42

Optional info
-------------

- intermittent failure (yes|no): no
- regression or new test:  new test

Failure output
--------------

```
[2019-06-03T16:35:03.713Z] ===============================================
[2019-06-03T16:35:03.713Z] Running test cmdLineTester_DataHelperTests_1 ...
[2019-06-03T16:35:03.713Z] ===============================================
[2019-06-03T16:35:03.713Z] cmdLineTester_DataHelperTests_1 Start Time: Mon Jun  3 13:35:02 2019 Epoch Time (ms): 1559579702739
[2019-06-03T16:35:03.713Z] "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/bin/java" -Xshareclasses:destroyAll; "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/bin/java" -Xshareclasses:groupAccess,destroyAll; echo "cache cleanup done";
[2019-06-03T16:35:03.713Z] JIT: env var TR_OptionsAOT is set to forceAOT
[2019-06-03T16:35:03.713Z] JIT: env var TR_Options is set to forceAOT
[2019-06-03T16:35:03.713Z] JVMSHRC005I No shared class caches available
[2019-06-03T16:35:03.713Z] JIT: env var TR_OptionsAOT is set to forceAOT
[2019-06-03T16:35:03.713Z] JIT: env var TR_Options is set to forceAOT
[2019-06-03T16:35:03.713Z] JVMSHRC005I No shared class caches available
[2019-06-03T16:35:03.713Z] cache cleanup done
[2019-06-03T16:35:03.713Z] variation: Mode610
[2019-06-03T16:35:03.713Z] JVM_OPTIONS: -Xcompressedrefs -Xcompressedrefs -Xjit -Xgcpolicy:gencon 
[2019-06-03T16:35:03.713Z] { itercnt=1; \
[2019-06-03T16:35:03.713Z] mkdir -p "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15595796638578/cmdLineTester_DataHelperTests_1_ITER_$itercnt"; \
[2019-06-03T16:35:03.713Z] cd "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15595796638578/cmdLineTester_DataHelperTests_1_ITER_$itercnt"; \
[2019-06-03T16:35:03.713Z] cp "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../../jvmtest/functional/cmdLineTests/shareClassTests/DataHelperTests/DataHelperTests.jar" .; \
[2019-06-03T16:35:03.713Z] "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/bin/jar" xf DataHelperTests.jar; \
[2019-06-03T16:35:03.713Z]  \
[2019-06-03T16:35:03.713Z] cd "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15595796638578/cmdLineTester_DataHelperTests_1/datacaching/dataone.contents"; "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/bin/jar" cf dataone.jar *.txt; cd "../../"; \
[2019-06-03T16:35:03.713Z] cd "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15595796638578/cmdLineTester_DataHelperTests_1/datacaching/datatwo.contents"; "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/bin/jar" cf datatwo.jar *.txt; cd "../../"; \
[2019-06-03T16:35:03.713Z] mv "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15595796638578/cmdLineTester_DataHelperTests_1/datacaching/dataone.contents/dataone.jar" "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15595796638578/cmdLineTester_DataHelperTests_1/datacaching"; \
[2019-06-03T16:35:03.713Z] mv "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15595796638578/cmdLineTester_DataHelperTests_1/datacaching/datatwo.contents/datatwo.jar" "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15595796638578/cmdLineTester_DataHelperTests_1/datacaching"; \
[2019-06-03T16:35:03.713Z] "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/bin/java" -Xcompressedrefs -Xcompressedrefs -Xjit -Xgcpolicy:gencon  -DJAVA_EXE='"/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/bin/java" -Xcompressedrefs -Xcompressedrefs -Xjit -Xgcpolicy:gencon ' -DCPDL=":" -DSCMODE=204 -DTEST_JVM_OPTIONS="-Xcompressedrefs -Xcompressedrefs -Xjit -Xgcpolicy:gencon " \
[2019-06-03T16:35:03.713Z] -jar "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../../jvmtest/functional/cmdline_options_tester/cmdlinetester.jar" \
[2019-06-03T16:35:03.713Z] -config "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../../jvmtest/functional/cmdLineTests/shareClassTests/DataHelperTests/DataHelperTests.xml" -xids all,11 -xlist "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../../jvmtest/functional/cmdLineTests/shareClassTests/DataHelperTests/exclude.xml" \
[2019-06-03T16:35:03.713Z] -nonZeroExitWhenError \
[2019-06-03T16:35:03.713Z] -outputLimit 300; \
[2019-06-03T16:35:03.713Z] if [ $? -eq 0 ] ; then echo ""; echo "cmdLineTester_DataHelperTests_1""_PASSED(ITER_$itercnt)"; echo ""; cd /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../..; rm -f -r "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15595796638578/cmdLineTester_DataHelperTests_1_ITER_$itercnt"; if [ $itercnt -eq 1 ] ; then echo ""; echo "cmdLineTester_DataHelperTests_1""_PASSED"; echo ""; fi else echo ""; echo "cmdLineTester_DataHelperTests_1""_FAILED(ITER_$itercnt)"; echo ""; echo ""; echo "cmdLineTester_DataHelperTests_1""_FAILED"; echo ""; exit 1; fi; } 2>&1 | tee -a "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15595796638578/TestTargetResult";
[2019-06-03T16:35:03.713Z] JIT: env var TR_OptionsAOT is set to forceAOT
[2019-06-03T16:35:03.713Z] JIT: env var TR_Options is set to forceAOT
[2019-06-03T16:35:04.700Z] /bin/sh: 7: cd: can't cd to /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15595796638578/cmdLineTester_DataHelperTests_1/datacaching/dataone.contents
[2019-06-03T16:35:04.700Z] JIT: env var TR_OptionsAOT is set to forceAOT
[2019-06-03T16:35:04.700Z] JIT: env var TR_Options is set to forceAOT
[2019-06-03T16:35:04.700Z] *.txt : no such file or directory
[2019-06-03T16:35:04.700Z] /bin/sh: 8: cd: can't cd to /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15595796638578/cmdLineTester_DataHelperTests_1/datacaching/datatwo.contents
[2019-06-03T16:35:04.700Z] JIT: env var TR_OptionsAOT is set to forceAOT
[2019-06-03T16:35:04.700Z] JIT: env var TR_Options is set to forceAOT
[2019-06-03T16:35:05.652Z] *.txt : no such file or directory
[2019-06-03T16:35:05.652Z] mv: cannot stat '/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15595796638578/cmdLineTester_DataHelperTests_1/datacaching/dataone.contents/dataone.jar': No such file or directory
[2019-06-03T16:35:05.652Z] mv: cannot stat '/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15595796638578/cmdLineTester_DataHelperTests_1/datacaching/datatwo.contents/datatwo.jar': No such file or directory
[2019-06-03T16:35:05.652Z] JIT: env var TR_OptionsAOT is set to forceAOT
[2019-06-03T16:35:05.652Z] JIT: env var TR_Options is set to forceAOT
[2019-06-03T16:35:05.652Z] Parsing exclude list...
[2019-06-03T16:35:05.652Z]   add excludes for [all]
[2019-06-03T16:35:05.652Z]   add excludes for [11]
[2019-06-03T16:35:05.652Z] DONE in 0 ms.
[2019-06-03T16:35:05.652Z] *** Starting test suite: Shared Classes DataCaching Helper API tests ***
[2019-06-03T16:35:05.652Z]  
[2019-06-03T16:35:05.652Z] #######################################################
[2019-06-03T16:35:05.652Z] Running tests with command line options: -Xshareclasses:reset,name=testCache
[2019-06-03T16:35:05.652Z]  
[2019-06-03T16:35:05.652Z] Executing command: "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/bin/java" -Xcompressedrefs -Xcompressedrefs -Xjit -Xgcpolicy:gencon  -version
[2019-06-03T16:35:06.678Z] 
[2019-06-03T16:35:06.678Z] Executing command: "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/bin/java" -Xcompressedrefs -Xcompressedrefs -Xjit -Xgcpolicy:gencon  -Xshareclasses:destroyAll
[2019-06-03T16:35:06.678Z] 
[2019-06-03T16:35:06.678Z] Testing: check the datacaching classloader is ok
[2019-06-03T16:35:06.678Z] Test start time: 2019/06/03 13:35:06 Atlantic Standard Time
[2019-06-03T16:35:06.678Z] Running command: "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/bin/java" -Xcompressedrefs -Xcompressedrefs -Xjit -Xgcpolicy:gencon  -Xshareclasses:reset,name=testCache   -classpath . apitesting.datahelper.DataCachingTest01
[2019-06-03T16:35:06.678Z] Time spent starting: 5 milliseconds
[2019-06-03T16:35:07.599Z] Time spent executing: 676 milliseconds
[2019-06-03T16:35:07.600Z] Test result: FAILED
[2019-06-03T16:35:07.600Z]  [ERR] JIT: env var TR_OptionsAOT is set to forceAOT
[2019-06-03T16:35:07.600Z]  [ERR] JIT: env var TR_Options is set to forceAOT
[2019-06-03T16:35:07.600Z]  [ERR] JVMSHRC023E Cache does not exist
[2019-06-03T16:35:07.600Z]  [ERR] Error: Could not find or load main class apitesting.datahelper.DataCachingTest01
[2019-06-03T16:35:07.600Z]  [ERR] Caused by: java.lang.ClassNotFoundException: apitesting.datahelper.DataCachingTest01
[2019-06-03T16:35:07.600Z] >> Success condition was not found: [Output match: test successful]
[2019-06-03T16:35:07.600Z] >> Failure condition was not found: [Output match: failed]
[2019-06-03T16:35:07.600Z] >> Failure condition was found: [Output match: exception:]
[2019-06-03T16:35:07.600Z] >> Failure condition was not found: [Output match: Processing dump event]
[2019-06-03T16:35:07.600Z] 
[2019-06-03T16:35:07.600Z] Testing: simple caching of a resource
[2019-06-03T16:35:07.600Z] Test start time: 2019/06/03 13:35:06 Atlantic Standard Time
[2019-06-03T16:35:07.600Z] Running command: "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/bin/java" -Xcompressedrefs -Xcompressedrefs -Xjit -Xgcpolicy:gencon  -Xshareclasses:reset,name=testCache -classpath . apitesting.datahelper.DataCachingTest02
[2019-06-03T16:35:07.600Z] Time spent starting: 6 milliseconds
[2019-06-03T16:35:08.519Z] Time spent executing: 660 milliseconds
[2019-06-03T16:35:08.519Z] Test result: FAILED
[2019-06-03T16:35:08.519Z]  [ERR] JIT: env var TR_OptionsAOT is set to forceAOT
[2019-06-03T16:35:08.519Z]  [ERR] JIT: env var TR_Options is set to forceAOT
[2019-06-03T16:35:08.519Z]  [ERR] JVMSHRC256I Persistent shared cache "testCache" has been destroyed
[2019-06-03T16:35:08.519Z]  [ERR] Error: Could not find or load main class apitesting.datahelper.DataCachingTest02
[2019-06-03T16:35:08.519Z]  [ERR] Caused by: java.lang.ClassNotFoundException: apitesting.datahelper.DataCachingTest02
[2019-06-03T16:35:08.519Z] >> Success condition was not found: [Output match: test successful]
[2019-06-03T16:35:08.519Z] >> Failure condition was not found: [Output match: failed]
[2019-06-03T16:35:08.519Z] >> Failure condition was found: [Output match: exception:]
[2019-06-03T16:35:08.519Z] >> Failure condition was not found: [Output match: Processing dump event]
[2019-06-03T16:35:08.519Z] 
...
```

--- a/test/functional/cmdLineTests/gcRegressionTests/playlist.xml
+++ b/test/functional/cmdLineTests/gcRegressionTests/playlist.xml
@@ -47,7 +47,7 @@
 		<variations>
 			<variation>Mode301</variation>
 		</variations>
-		<command>$(JAVA_COMMAND) $(JVM_OPTIONS) -DTESTSJARPATH=$(Q)$(TEST_RESROOT)$(D)gcRegressionTests.jar$(Q) -DRESJAR=$(CMDLINETESTER_RESJAR) -DTESTOUTPUT=$(TESTOUTPUT) \
+		<command>$(JAVA_COMMAND) $(JVM_OPTIONS) -DTESTSJARPATH=$(Q)$(TEST_RESROOT)$(D)gcRegressionTests.jar$(Q) -DRESJAR=$(CMDLINETESTER_RESJAR) \
 		-DEXE=$(SQ)$(JAVA_COMMAND) $(JVM_OPTIONS)$(SQ) -Xint -jar $(CMDLINETESTER_JAR) -config $(Q)$(TEST_RESROOT)$(D)gcRegressionTests.xml$(Q) \
 		-verbose -explainExcludes -xids all,$(PLATFORM),Mode301 -plats all,$(PLATFORM),Mode301 -xlist $(Q)$(TEST_RESROOT)$(D)gcRegressionTests_excludes.xml$(Q) -explainExcludes -nonZeroExitWhenError; \
 		$(TEST_STATUS)</command>

--- a/test/functional/cmdLineTests/shareClassTests/DataHelperTests/playlist.xml
+++ b/test/functional/cmdLineTests/shareClassTests/DataHelperTests/playlist.xml
@@ -33,10 +33,10 @@
 		<command>cp $(Q)$(TEST_RESROOT)$(D)DataHelperTests.jar$(Q) .; \
 	$(Q)$(TEST_JDK_HOME)$(D)bin$(D)jar$(EXECUTABLE_SUFFIX)$(Q) xf DataHelperTests.jar; \
 	$(CONVERT_TO_EBCDIC_CMD) \
-	$(CD) $(Q)$(TESTOUTPUT)$(D)$@$(D)datacaching$(D)dataone.contents$(Q); $(Q)$(TEST_JDK_HOME)$(D)bin$(D)jar$(EXECUTABLE_SUFFIX)$(Q) cf dataone.jar *.txt; $(CD) $(Q)..$(D)..$(D)$(Q); \
-	$(CD) $(Q)$(TESTOUTPUT)$(D)$@$(D)datacaching$(D)datatwo.contents$(Q); $(Q)$(TEST_JDK_HOME)$(D)bin$(D)jar$(EXECUTABLE_SUFFIX)$(Q) cf datatwo.jar *.txt; $(CD) $(Q)..$(D)..$(D)$(Q); \
-	mv $(Q)$(TESTOUTPUT)$(D)$@$(D)datacaching$(D)dataone.contents$(D)dataone.jar$(Q) $(Q)$(TESTOUTPUT)$(D)$@$(D)datacaching$(Q); \
-	mv $(Q)$(TESTOUTPUT)$(D)$@$(D)datacaching$(D)datatwo.contents$(D)datatwo.jar$(Q) $(Q)$(TESTOUTPUT)$(D)$@$(D)datacaching$(Q); \
+	$(CD) $(Q)$(REPORTDIR)$(D)datacaching$(D)dataone.contents$(Q); $(Q)$(TEST_JDK_HOME)$(D)bin$(D)jar$(EXECUTABLE_SUFFIX)$(Q) cf dataone.jar *.txt; $(CD) $(Q)..$(D)..$(D)$(Q); \
+	$(CD) $(Q)$(REPORTDIR)$(D)datacaching$(D)datatwo.contents$(Q); $(Q)$(TEST_JDK_HOME)$(D)bin$(D)jar$(EXECUTABLE_SUFFIX)$(Q) cf datatwo.jar *.txt; $(CD) $(Q)..$(D)..$(D)$(Q); \
+	mv $(Q)$(REPORTDIR)$(D)datacaching$(D)dataone.contents$(D)dataone.jar$(Q) $(Q)$(REPORTDIR)$(D)datacaching$(Q); \
+	mv $(Q)$(REPORTDIR)$(D)datacaching$(D)datatwo.contents$(D)datatwo.jar$(Q) $(Q)$(REPORTDIR)$(D)datacaching$(Q); \
 	$(JAVA_COMMAND) $(JVM_OPTIONS) -DJAVA_EXE=$(SQ)$(JAVA_COMMAND) $(JVM_OPTIONS)$(SQ) -DCPDL=$(Q)$(P)$(Q) -DSCMODE=204 -DTEST_JVM_OPTIONS=$(Q)$(JVM_OPTIONS)$(Q) \
 	-jar $(CMDLINETESTER_JAR) \
 	-config $(Q)$(TEST_RESROOT)$(D)DataHelperTests$(XMLSUFFIX).xml$(Q) -xids all,$(JDK_VERSION) -xlist $(Q)$(TEST_RESROOT)$(D)exclude.xml$(Q) \

--- a/test/functional/cmdLineTests/shareClassTests/SCCMLTests/playlist.xml
+++ b/test/functional/cmdLineTests/shareClassTests/SCCMLTests/playlist.xml
@@ -123,14 +123,14 @@
 	true DataHelperTests.jar ;\
 	$(Q)$(TEST_JDK_HOME)$(D)bin$(D)jar$(EXECUTABLE_SUFFIX)$(Q) xf DataHelperTests.jar; \
 	$(CONVERT_TO_EBCDIC_CMD) \
-	$(CD) $(Q)$(TESTOUTPUT)$(D)$@$(D)datacaching$(D)dataone.contents$(Q); \
+	$(CD) $(Q)$(REPORTDIR)$(D)datacaching$(D)dataone.contents$(Q); \
 	$(Q)$(TEST_JDK_HOME)$(D)bin$(D)jar$(EXECUTABLE_SUFFIX)$(Q) cf dataone.jar *.txt; \
 	$(CD) $(REPORTDIR); \
-	$(CD) $(Q)$(TESTOUTPUT)$(D)$@$(D)datacaching$(D)datatwo.contents$(Q); \
+	$(CD) $(Q)$(REPORTDIR)$(D)datacaching$(D)datatwo.contents$(Q); \
 	$(Q)$(TEST_JDK_HOME)$(D)bin$(D)jar$(EXECUTABLE_SUFFIX)$(Q) cf datatwo.jar *.txt; \
 	$(CD) $(REPORTDIR); \
-	mv $(Q)$(TESTOUTPUT)$(D)$@$(D)datacaching$(D)dataone.contents$(D)dataone.jar$(Q) $(Q)$(TESTOUTPUT)$(D)$@$(D)datacaching$(Q); \
-	mv $(Q)$(TESTOUTPUT)$(D)$@$(D)datacaching$(D)datatwo.contents$(D)datatwo.jar$(Q) $(Q)$(TESTOUTPUT)$(D)$@$(D)datacaching$(Q); \
+	mv $(Q)$(REPORTDIR)$(D)datacaching$(D)dataone.contents$(D)dataone.jar$(Q) $(Q)$(REPORTDIR)$(D)datacaching$(Q); \
+	mv $(Q)$(REPORTDIR)$(D)datacaching$(D)datatwo.contents$(D)datatwo.jar$(Q) $(Q)$(REPORTDIR)$(D)datacaching$(Q); \
 	$(JAVA_COMMAND) $(JVM_OPTIONS) \
 	-DPATHSEP=$(Q)$(D)$(Q) -DCPDL=$(Q)$(P)$(Q) -DRUN_SCRIPT=$(RUN_SCRIPT) -DPROPS_DIR=$(PROPS_DIR) -DSCRIPT_SUFFIX=$(SCRIPT_SUFFIX) -DEXECUTABLE_SUFFIX=$(EXECUTABLE_SUFFIX) \
 	-DJAVA_EXE=$(SQ)$(JAVA_COMMAND) $(JVM_OPTIONS)$(SQ) -DJAVA_HOME=$(SQ)$(TEST_JDK_HOME)$(SQ) -DSCMODE=204 -DJVM_TEST_ROOT=$(Q)$(JVM_TEST_ROOT)$(Q) \

